File: 0049-group-anagrams/0049-group-anagrams.cpp
#include <algorithm>
#include <unordered_map>
#include <vector>
#include <string>

class Solution {
public:
  std::vector<std::vector<std::string>> groupAnagrams(std::vector<std::string> &strs) {
    std::unordered_map<std::string, std::vector<std::string>> mp;;

    for (std::string s : strs){
      std::string t = s;
      std::sort(t.begin(), t.end());

      mp[t].push_back(s);
    }

    std::vector<std::vector<std::string>> answer;
    for(std::pair<std::string, std::vector<std::string>> group: mp){
      answer.push_back(group.second);
    }

    return answer;
  }
};

File: 0049-group-anagrams/0049-group-anagrams.py
from typing import List

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        groups: List[List [str]] = []
        hash_map= {}

        for word in strs:
            sorted_word = "".join(sorted(word))
            if sorted_word in hash_map:
                hash_map[sorted_word].append(word)
            else:
                hash_map[sorted_word] = [word]

        for group in hash_map.values():
            groups.append(group)

        return groups
